code/datums/flock/flock.dm

```diff
@@ -392,6 +392,40 @@
 			if(C)
 				C.images -= I
 
+// naming
+
+/datum/flock/proc/pick_name(flock_type)
+	var/name
+	switch(flock_type)
+		if ("flock")
```

Comment:
Do we really need to do it by type? If anything it'd probably be better if each one was globally unique, rather than just unique-for-drones or unique-for-traces

Reply:
Confused, do you mean unique across flocks?

Reply:
I just mean it's probably better to get rid of the switch, and just do 
```
for(var/mob/living/F in mobs)
if(isflock(F))
if(F.name == name)
found = TRUE
break
```

So a given flock entity is guaranteed to have a unique name in the game, not just a unique name for all the flockdrones in that flock. Hit me up on discord if it's not clear.

Reply:
Yeah, that make sense, maybe sounds a bit more of a design decision that we should talk about though?

Reply:
After discord discussion it turns out I'm blind. Consider this resolved. but maybe also consider doing the istype() thing instead of strings